client/core/core.go

```diff
@@ -1046,8 +1047,42 @@ func (c *Core) Run(ctx context.Context) {
 	stopDB()
 	dbWG.Wait()
 
+	// At this point, it should be safe to access the data structures without
+	// mutex protection. Goroutines have returned, and consumers should not call
+	// Core methods after shutdown. We'll play it safe anyway.
```

Comment:
> Goroutines have returned

I'm also seeing some routes like `Login`, `initializeDEXConnections`,  `checkUnpaidFees` that either are or start unmonitored goroutines  that don't use the `Core`'s `WaitGroup`. Not sure how problematic they are for this assertion though.

Reply:
True, `checkUnpaidFees` is a wild one, and I'd like to see it go away soon anyway, with the new registration protocol.  I've kept the `conns` and `trades` map accesses safe, since a concurrent map write would cause a panic, unlike with a regular variable.

Consumers really should not cancel until all calls to exported Core methods are returned, but that may not be possible if something is hung, so let's consider this comment a design goal.  I think the redeem search goroutine might also be wild.